core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java

```diff
@@ -4659,6 +4659,49 @@ private void checkNullOperand(SqlTester tester, String op) {
     tester.checkNull("json_depth(cast(null as varchar))");
   }
 
+  @Test public void testJsonInsert() {
+    tester.checkString("json_insert('10', '$.a', 10, '$.c', '[true]')",
+            "10", "VARCHAR(2000) NOT NULL");
+    tester.checkString("json_insert('{ \"a\": 1, \"b\": [2]}', '$.a', 10, '$.c', '[true]')",
+            "{\"a\":1,\"b\":[2],\"c\":\"[true]\"}", "VARCHAR(2000) NOT NULL");
+    tester.checkString("json_insert('{ \"a\": 1, \"b\": [2]}', '$', 10, '$', '[true]')",
+            "{\"a\":1,\"b\":[2]}", "VARCHAR(2000) NOT NULL");
+    tester.checkString("json_insert('{ \"a\": 1, \"b\": [2]}', '$.b[1]', 10)",
+            "{\"a\":1,\"b\":[2,10]}", "VARCHAR(2000) NOT NULL");
+    tester.checkString("json_insert('{\"a\": 1, \"b\": [2, 3, [true]]}', '$.b[3]', 'false')",
+            "{\"a\":1,\"b\":[2,3,[true],\"false\"]}", "VARCHAR(2000) NOT NULL");
+    tester.checkFails("json_insert('{\"a\": 1, \"b\": [2, 3, [true]]}', 'a', 'false')",
+            "(?s).*Not a valid input for.*", true);
```

Comment:
Please add some null checks as well as we have done for other methods (the checks are now always located at the tail of test method body).